### DISCUSSION TITLE
I have 3 questions regarding customising theme colors, customising Sidebar and Google OAuth Provider in 4.x versions.

### DISCUSSION THREAD

**Discussion by kotharismeet-3** [refinedev/refine#4647] [Q&A] [answered]
1. How to customise colors, I tried following!
Tried changing ColorModeContextProvider > RefineThemes > LightTheme > lightPalletes and darkPalletes same too. But UI is not changing
2. How to customise icons in sidebar in `ThemeSiderV2` ?
3. How to pass Google Client Id in provider to AuthPage ?

**Comment by Ashwin-op** [comment]
2. You can pass a custom icon in `meta.icon` in the `resources` parameter of `Refine` component

```ts
...
  <Refine
    ...
    resources={[
      ...
      {
        ...
        meta: {
          ...
          icon: <CustomIcon />,
        },
      },
    ]}
    ...
  >
    ...
  </Refine>
...
```

**Comment by alicanerdurmaz** [comment, marked as answer]
1. You can follow this documentation for theme switch. If you have any problem please provide a reproducible example for us to debug.
Material UI: https://refine.dev/docs/api-reference/mui/theming/#dark-mode-toggle
Mantine: https://refine.dev/docs/api-reference/mantine/theming/#theme-switching
Chakra UI: https://refine.dev/docs/api-reference/chakra-ui/theming/#theme-switching
Ant Design: https://refine.dev/docs/api-reference/antd/theming/#switching-to-dark-theme

3. When you click the provider button from `<AuthPage/>`, it calls `authProvider.login` function with the `providerName` param. you need to handle Google login logic inside`authProvider`

**Reply by liushooter** [reply]
Thank your answer.

Now，I support 2 login methods: email and github OAuth.

How to solve the problem？


 <img width="250px" src="https://github.com/refinedev/refine/assets/1235106/691b4729-17d0-46a8-9088-53237d45627f">

For more details: https://github.com/refinedev/refine/discussions/4836
